src/protocols/mp_services.rs

```diff
@@ -0,0 +1,122 @@
+//! MP Services Protocol
+//!
+//! This Protocol is defined in UEFI Platform Integration Specification, Section 13.4
+//!
+//! The MP Services Protocol provides a generalized way of performing following tasks:
+//! - Retrieving information of multi-processor environment and MP-related status of specific processors.
+//! - Dispatching user-provided function to APs.
+//! - Maintain MP-related processor status.
+
+pub const PROTOCOL_GUID: crate::base::Guid = crate::base::Guid::from_fields(
+    0x3fdda605,
+    0xa76e,
+    0x4f46,
+    0xad,
+    0x29,
+    &[0x12, 0xf4, 0x53, 0x1b, 0x3d, 0x08],
+);
+
+pub type StatusFlag = u32;
+
+pub const PROCESSOR_AS_BSP_BIT: StatusFlag = 0x00000001;
+pub const PROCESSOR_ENABLED_BIT: StatusFlag = 0x00000002;
+pub const PROCESSOR_HEALTH_STATUS_BIT: StatusFlag = 0x00000004;
+
+pub const END_OF_CPU_LIST: usize = 0xffffffff;
```

Comment:
This is tricky. I changed it to `usize::MAX`. The spec defines this as a preprocessor macro, and thus depending on context it is very likely interpreted as an `int`-literal, and thus subject to sign-extension on assignment. The spec-wording is vague, but I think the correct conversion is a `usize` as you suggested, but then with all bits set.

I am unsure, but I think your code would result in `u32::MAX` if `usize` is 64bit, instead of `u64::MAX`.